velero-csi-plugin/csi_snapshotter.go

```diff
@@ -149,10 +143,17 @@ func (p *CSISnapshotter) Execute(item runtime.Unstructured, backup *velerov1api.
 	if annotations == nil {
 		annotations = make(map[string]string)
 	}
-	annotations[veleroSnapshotNameAnnotation] = upd.Name
+	annotations[volumeSnapshotLabel] = upd.Name
 	annotations[velerov1api.BackupNameLabel] = backup.Name
 	accessor.SetAnnotations(annotations)
 
+	labels := accessor.GetLabels()
```

Comment:
the `accessor` is probably unnecessary here too, but not blocking, we can clean it up more later

Reply:
I was going to clean this up in a follow-up PR. But... 😄 
Refactored and added some tests

Reply:
The `accessor` here is necessary as the annotations and labels need to be persisted.
We can refactor and clean up in a later PR

Reply:
Yep -- like I said, not blocking, so fine to address in a separate PR.

To be clear on what I mean -- we're already converting `item` to a typed PVC object at https://github.com/vmware-tanzu/velero-plugin-for-csi/pull/18/files#diff-f8c4865ab47acc13ac1a532be0ba9e0dR66-R69, so what I would do is make any modification you need to `pvc.Annotations`, `pvc.Labels` etc, then at the bottom of this function, convert `pvc` back to an Unstructured and return it, rather than returning `item` directly.

It's not a huge deal either way, but that's more consistent with other plugins, and we're already converting `item` (Unstructured) to `pvc` (typed PVC), so may as well use `pvc`.

Reply:
See, for example, https://github.com/vmware-tanzu/velero/blob/master/pkg/restore/service_action.go#L48-L68, where the flow is:

1. convert the unstructured input to a typed Service object
2. manipulate the typed Service object
3. convert the Service object back to an unstructured representation
4. return the unstructured

Reply:
will do this in a follow-up